src/libcork/posix/files.c

```diff
@@ -131,8 +131,9 @@ cork_path_set_absolute(struct cork_path *path)
     struct cork_buffer  buf;
 
     if (path->given.size > 0 &&
-        cork_buffer_char(&path->given, path->given.size - 1) == '/') {
-        /* The path is already absolute */
+        (cork_buffer_char(&path->given, 0) == '/') &&
+        (cork_buffer_char(&path->given, path->given.size - 1) == '/')) {
```

Comment:
Yes, too strict — you don't have to have a trailing slash for a path to be absolute.  (`/home/dcreager` is absolute.)